server/db/UnoDB.go

```diff
@@ -0,0 +1,33 @@
+package db
+
+import (
+	"github.com/jak103/uno/model"
+)
+
+// UnoDB declares the database types for the applicaiton
+type UnoDB interface {
```

Comment:
As I understand it, this interface needs to have provisions for replacing all state updates from the `server/uno.go` file. This would mean we need ways to retrieve and modify a players hand for a given game, at the least. None of the existing interfaces appear to provide the needed functionality...

Reply:
So.. to clarify my thoughts behind this. And this is where @hhenrichsen have had a difference of opinion.  I personally felt like the purpose of this task was to implement the `structure` in order to have an interface for different database connection types.  (ie, FireStore, Mongo, MockDB)

I personally didn't feel like the scope of this Issue was to actually `implement` all of the particular game functionality. There were other issues logged to ensure that can happen.

This particular interface can absolutely be expanded on by other team members. But If they don't know that the interface is there, or in place.... they won't add to it.  My concern is that we could spend all week rounding out all of the database reqeusts... and every other group working on a task will implement a database calls with some other paradigm in mind.  

TL;DR I agree that this interface doesn't implement all of the needed endpoints.... I just don't feel that was within the scope of this Task.. Which was in my opinion... to just implement the `stucture` of the database connection pattern.

Reply:
In my mind, the DB interface should just be able to CRUD the game data. The uno.go code retrieves the game object, manipulates it and then stores it again. If we do as you are suggesting Aaron, then all the game functionality move into the database interface, which is not what we want.

Reply:
I also see from the interface that it can also CRUD the player data. This is something I hadn't thought of, but is correct. We probably want to decouple the player and game data as we will want the player to exist beyond the scope of the game.
This will allow us to track stats and other info for the players. I like it.